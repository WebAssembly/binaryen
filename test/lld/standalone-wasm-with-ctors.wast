(module
 (memory $0 2)
 (table $0 1 1 funcref)
 (elem (i32.const 0) $foo)
 (global $global$0 (mut i32) (i32.const 66112))
 (global $global$1 i32 (i32.const 66112))
 (global $global$2 i32 (i32.const 576))
 (export "memory" (memory $0))
 (export "__heap_base" (global $global$1))
 (export "__data_end" (global $global$2))
 (export "__wasm_call_ctors" (func $__wasm_call_ctors))
 (func $foo (result i32))
 (func $__wasm_call_ctors
  (nop)
 )
)

