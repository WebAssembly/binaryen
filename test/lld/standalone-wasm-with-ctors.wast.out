(module
 (type $FUNCSIG$i (func (result i32)))
 (type $FUNCSIG$v (func))
 (memory $0 2)
 (table $0 1 1 funcref)
 (elem (i32.const 0) $foo)
 (global $global$0 (mut i32) (i32.const 66112))
 (global $global$1 i32 (i32.const 66112))
 (global $global$2 i32 (i32.const 576))
 (export "memory" (memory $0))
 (export "__heap_base" (global $global$1))
 (export "__data_end" (global $global$2))
 (export "__wasm_call_ctors" (func $__wasm_call_ctors))
 (export "stackSave" (func $stackSave))
 (export "stackAlloc" (func $stackAlloc))
 (export "stackRestore" (func $stackRestore))
 (export "__growWasmMemory" (func $__growWasmMemory))
 (func $foo (; 0 ;) (type $FUNCSIG$i) (result i32)
  (nop)
 )
 (func $__wasm_call_ctors (; 1 ;) (type $FUNCSIG$v)
  (nop)
 )
 (func $stackSave (; 2 ;) (result i32)
  (global.get $global$0)
 )
 (func $stackAlloc (; 3 ;) (param $0 i32) (result i32)
  (local $1 i32)
  (global.set $global$0
   (local.tee $1
    (i32.and
     (i32.sub
      (global.get $global$0)
      (local.get $0)
     )
     (i32.const -16)
    )
   )
  )
  (local.get $1)
 )
 (func $stackRestore (; 4 ;) (param $0 i32)
  (global.set $global$0
   (local.get $0)
  )
 )
 (func $__growWasmMemory (; 5 ;) (param $newSize i32) (result i32)
  (memory.grow
   (local.get $newSize)
  )
 )
)
(;
--BEGIN METADATA --
{
  "staticBump": 8,
  "tableSize": 1,
  "declares": [
  ],
  "externs": [
  ],
  "implementedFunctions": [
    "___wasm_call_ctors",
    "_stackSave",
    "_stackAlloc",
    "_stackRestore",
    "___growWasmMemory"
  ],
  "exports": [
    "__wasm_call_ctors",
    "stackSave",
    "stackAlloc",
    "stackRestore",
    "__growWasmMemory"
  ],
  "namedGlobals": {
    "__heap_base" : "66112",
    "__data_end" : "576"
  },
  "invokeFuncs": [
  ],
  "features": [
  ],
  "mainReadsParams": 0
}
-- END METADATA --
;)
